proto/astarte_msg_hub.proto

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is part of Astarte.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+syntax = "proto3";
+
+package org.astarteplatform.msghub;
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+
+service MessageHub {
+  rpc Attach(Node) returns (stream NodeMessage) {}
+  rpc Send(NodeMessage) returns (google.protobuf.Empty){}
+  rpc Detach(Node) returns (google.protobuf.Empty){}
+}
+
+message Node {
+  string id = 1;
+  NodeIntrospection introspection = 2;
+}
+
+message NodeIntrospection {
+  repeated Interface interfaces = 1;
+}
+
+message Interface {
+  string name = 1;
+  int32 major = 2;
+  int32 minor = 3;
+}
+
+message NodeMessage{
+  Interface interface = 1;
+  string path = 2;
+  AstarteAggregation astarte_aggregation = 3;
+  AstarteDataType data_type = 4;
+}
+
+enum AstarteAggregation {
+  INDIVIDUAL = 0;
+  OBJECT = 1;
+}
+
+message AstarteDataType {
+  oneof dataType {
+    double double = 1;
+    int32 int32 = 2;
+    int64 int64 = 3;
+    bool bool = 4;
+    string string = 5;
+    bytes bytes = 6;
+    google.protobuf.Timestamp time = 7;
+    AstarteDataArrayType data_array = 8;
+    AstarteDataObjectType data_map = 9;
+  }
+}
+
+message AstarteDataArrayType {
+  repeated AstarteDataType array_type = 1;
+}
+
+message AstarteDataObjectType {
+  map<string, AstarteDataType> data_map= 1;
+}
```

Comment:
I think we should work with a different type hierarchy:
I suggest taking a look to https://github.com/astarte-platform/astarte_flow/blob/2b870392d4cf7dff7e41aa3529da53237d7eaa76/lib/astarte_flow/message.ex#L45

But basically:
- basic types (scalars)
- arrays of basic types
- objects of arrays or basic types